inflect/__init__.py

```diff
@@ -3839,23 +3839,28 @@ def number_to_words(  # noqa: C901
     ) -> Union[str, List[str]]:
         """
         Return a number in words.
-
         group = 1, 2 or 3 to group numbers before turning into words
         comma: define comma
-
         andword:
             word for 'and'. Can be set to ''.
             e.g. "one hundred and one" vs "one hundred one"
-
         zero: word for '0'
         one: word for '1'
         decimal: word for decimal point
         threshold: numbers above threshold not turned into words
-
         parameters not remembered from last call. Departure from Perl version.
         """
         self._number_args = {"andword": andword, "zero": zero, "one": one}
-        num = str(num)
+
+        # Handle scientific notation conversion
+        if isinstance(num, float):
+            formatted = f"{num:.17g}"
+            if 'e' in formatted.lower():
+                num = f"{num:.17f}".rstrip('0').rstrip('.')
+            else:
+                num = formatted
+        else:
+            num = str(num)
```

Comment:
I'd recommend to extract this logic into a helper function/method, something like `num = _normalize_number(num)`. That way, the comment can be in the docstring. This approach also limits the cyclomatic complexity.